content/blog/2024-01-30-flash-framework.Rmd

```diff
@@ -0,0 +1,67 @@
+
+
+<center>
+![**Fig 1** Standard Approach for Alerting Systems](/blog/2024-01-30-flash-framework/image3.png)</center>
+
+1. Predict a value that you expect the data to look like 
+2. Calculate the difference between your observed and predicted values 
+3. Contextualize the difference (e.g., to account for the variability in the data)
+4. Send alerts when this value passes a threshold
+
+Using this formula, if we assume that our data follows a bell curve, we can calculate the difference between our prediction (which might be the average of the data) and what we observe and divide by the standard deviation of the data. If this value exceeds a specific threshold of 3, then the probability that we would observe that data point in that bell curve would be less than 1%, which means that it is quite surprising. By setting this threshold high, only a small percentage of points are identified as outliers and result in alerts.
+
+This was the same intuition we used while initially designing Delphi's data monitoring system at the pandemic's start. This monitoring system was particularly important because Delphi's stakeholders wanted to be aware of any events in the data - like changes in data quality or disease dynamics. First, we split these events into data validation (e.g., negative cases) and events that needed a human-facing statistical alert (using the above formula) so that an expert could ultimately interpret and classify the event. 
+
+However, we quickly realized this formula did not work with Delphi's large data volume. To illustrate, creating alerts for 1% of the data over millions of data points, like those analyzed by Delphi, results in 10,000+ alerts - far too many for us at Delphi to inspect and investigate! Standard correction approaches were also incompatible with public health data. Our first attempt was reducing the percentage of alerts (e.g., thresholds that would identify from 1% of the data as alerts to 0.01%). However, because modeling public health data is difficult, defining thresholds (e.g., 1 in 10000) when data streams only have a few hundred historical data points can be somewhat arbitrary. Additionally, when applying the same standard threshold-generating function across all the data (adaptive thresholds), we missed some important events from signals that were not well-calibrated to the function. Tuning these parameters by hand across hundreds of signals was tedious, and Delphi turned off the outlier alerting system. 
```

Comment:
nit:
```suggestion
However, we quickly realized this formula did not work with Delphi's large data volume. To illustrate: creating alerts for 1% of the data over millions of data points, like those analyzed by Delphi, results in 10,000+ alerts - far too many for us at Delphi to inspect and investigate! Standard correction approaches were also incompatible with public health data. Our first attempt was reducing the percentage of alerts (e.g., changing thresholds that would identify from 1% of the data as alerts to 0.01%). However, because modeling public health data is difficult, defining thresholds (e.g., 1 in 10000) when data streams only have a few hundred historical data points can be somewhat arbitrary. Additionally, when applying the same standard threshold-generating function across all the data (adaptive thresholds), we missed some important events from signals that were not well-calibrated to the function. Tuning these parameters by hand across hundreds of signals was tedious, and Delphi turned off the outlier alerting system. 
```